src/strings/unicode_ops.c

```diff
@@ -400,3 +440,42 @@ void MVM_unicode_release(MVMThreadContext *tc)
     }
     uv_mutex_unlock(&property_hash_count_mutex);
 }
+/* Looks up a grapheme by name. Lazily constructs a hash. */
+MVMString * MVM_unicode_string_from_name(MVMThreadContext *tc, MVMString *name) {
+    MVMuint64 size;
+    char *cname = MVM_string_ascii_encode(tc, name, &size, 0);
+    MVMUnicodeGraphemeNameRegistry *result;
+
+    MVMGrapheme32 result_graph;
+    const MVMint32 * uni_seq;
+    int array_size;
+    MVMint32 cp;
+    MVMNormalizer norm;
+
+    result_graph = MVM_unicode_lookup_by_name(tc, name);
+    /* If it's just a codepoint, return that */
+    if (result_graph >= 0)
+        return MVM_string_from_grapheme(tc, result_graph);
+
+    if (!property_codes_by_seq_names) {
+        generate_property_codes_by_seq_names(tc);
+    }
+    HASH_FIND(hash_handle, property_codes_by_seq_names, cname, strlen((const char *)cname), result);
+    MVM_free(cname);
+    /* If we can't find a result return an empty string */
+    if (!result)
+        return tc->instance->str_consts.empty;
+
+    uni_seq = uni_seq_enum[result->structitem];
+    /* The first element is the number of codepoints in the sequence */
+    array_size = uni_seq[0];
+    MVM_unicode_normalizer_init(tc, &norm, MVM_NORMALIZE_NFG);
+    for (int i = 1; i < array_size; i++) {
+        cp = uni_seq[i];
+        MVM_unicode_normalizer_process_codepoint_full(tc, &norm, cp, &cp);
+    }
+    MVM_unicode_normalizer_process_codepoint_norm_terminator(tc, &norm, cp, &result_graph);
```

Comment:
I don't understand `cp` being fed into the normalizer here; either it's what we already put in, or it's whatever was already written into `cp` by the normalizer the last time around the loop. Should this just be calling the `eof`` function?